.github/workflows/stackhpc-container-image-build.yml

```diff
@@ -200,18 +207,58 @@ jobs:
       - name: Fail if no images have been built
         run: if [ $(wc -l < ${{ matrix.distro }}-container-images) -le 1 ]; then exit 1; fi
 
-      - name: Upload container images artifact
+      - name: Scan built container images
+        run: src/kayobe-config/tools/scan-images.sh ${{ matrix.distro }} ${{ needs.generate-tag.outputs.kolla_tag }}
+
+      - name: Upload Trivy scan results artifact
         uses: actions/upload-artifact@v4
         with:
-          name: ${{ matrix.distro }} container images
-          path: ${{ matrix.distro }}-container-images
+          name: ${{ matrix.distro }}-image-scan-output
+          path: image-scan-output
           retention-days: 7
 
+      - name: Fail if no images have passed scanning
+        run: if [ $(wc -l < image-scan-output/clean-images.txt) -le 0 ]; then exit 1; fi
+        if: github.event.inputs.scan-push == 'false'
+
+      - name: Append dirty images to clean list
+        run: |
+          cat image-scan-output/dirty-images.txt >> image-scan-output/clean-images.txt
+        if: github.event.inputs.scan-push == 'true'
+
+      - name: Push images
+        run: |
+          source venvs/kayobe/bin/activate &&
+          source src/kayobe-config/kayobe-env --environment ci-builder &&
+          kayobe playbook run ${KAYOBE_CONFIG_PATH}/ansible/docker-registry-login.yml &&
+
+          while read -r image; do
+            # Retries!
+            for i in {1..10}; do 
+              docker push ${image} && break || sleep 5
```

Comment:
It might be helpful to have some logging about the attempt number etc. here. We should also fail if the final attempt fails.

I think the default number of retries in kolla is 3, and I don't see any override. How about 5?

Reply:
Done